router/pkg/metric/metric_store.go

```diff
@@ -74,28 +74,34 @@ type (
 		otelMeterProvider *metric.MeterProvider
 		promMeterProvider *metric.MeterProvider
 
-		processStartTime time.Time
-
+		processStartTime   time.Time
 		otlpRequestMetrics Provider
 		promRequestMetrics Provider
 
 		logger *zap.Logger
 	}
 
 	Provider interface {
-		MeasureInFlight(ctx context.Context, attr ...attribute.KeyValue) func()
-		MeasureRequestCount(ctx context.Context, attr ...attribute.KeyValue)
-		MeasureRequestSize(ctx context.Context, contentLength int64, attr ...attribute.KeyValue)
-		MeasureResponseSize(ctx context.Context, size int64, attr ...attribute.KeyValue)
-		MeasureLatency(ctx context.Context, latency time.Duration, attr ...attribute.KeyValue)
-		MeasureRequestError(ctx context.Context, attr ...attribute.KeyValue)
-		MeasureOperationPlanningTime(ctx context.Context, planningTime time.Duration, attr ...attribute.KeyValue)
+		MeasureInFlight(ctx context.Context, opts ...otelmetric.AddOption) func()
+		MeasureRequestCount(ctx context.Context, opts ...otelmetric.AddOption)
+		MeasureRequestSize(ctx context.Context, contentLength int64, opts ...otelmetric.AddOption)
+		MeasureResponseSize(ctx context.Context, size int64, opts ...otelmetric.AddOption)
+		MeasureLatency(ctx context.Context, latency float64, opts ...otelmetric.RecordOption)
+		MeasureRequestError(ctx context.Context, opts ...otelmetric.AddOption)
+		MeasureOperationPlanningTime(ctx context.Context, planningTime float64, opts ...otelmetric.RecordOption)
 		Flush(ctx context.Context) error
 	}
 
 	Store interface {
+		MeasureInFlight(ctx context.Context, sliceAttr, attr []attribute.KeyValue) func()
```

Comment:
Can you add godocs here for how the interfaces are used, and how they are coupled?

Reply:
Done